lib/broadway.ex

```diff
@@ -129,6 +129,88 @@ defmodule Broadway do
   or `c:handle_batch/4`. This information will be sent back to
   the producer which will correctly acknowledge the message.
 
+  ## Working with standard GenStage producers
```

Comment:
This documentation is getting a bit long... do you think it is worth moving this to Broadway.Producer but keep the rest of the Broadway.Producer API private for now?

Reply:
I think we could move "General Architecture" to its own page before moving this one. I could also do it in a separated PR among with other changes. Maybe also creating a "Getting started" page. WDYT?

Reply:
Sounds good. The only note is that I believe Broadway.Producer should be usable outside of Broadway, so we will have to expose it as a public API. That's why I think moving it there is a good candidate.